src/ubx.cpp

```diff
@@ -1664,6 +1666,23 @@ GPSDriverUBX::payloadRxDone()
 		ret = 1;
 		break;
 
+	case UBX_MSG_TIM_TIM2:
+		UBX_TRACE_RXMSG("Rx TIM-TIM2");
+
+		{
+			// convert to unix timestamp
+			uint64_t tv_sec = _buf.payload_rx_tim_tim2.wnR * 7 * 24 * 3600;
```

Comment:
Just note: If we need add 1980.01.06(gps start time)?

Reply:
Now just based on the data sheet, I can't confirm if it need to add.

Reply:
Can you check that? According to the datasheet it depends on the `flags` bits.